toolkit/docs/how_it_works/3_package_building.md

```diff
@@ -181,42 +171,22 @@ Once the packages are cached they are copied into `./../out/RPMs` for use in fur
+### Stage 3: Scheduler
+The `scheduler` tool's job is to walk the dependency graph and build every specified package and their dependencies.
 
-`Unravel` implements a modular setup which can output different formats. The most basic format is a simple topologically sorted list of packages which can be build sequentially. Only `TypeBuild` nodes need to be printed as all other node types are simply to enforce ordering in the grpah.
-
-The more useful system, and what is used in further steps by the build system, is a recursive `Makefile`. Since the build system is already using `Make` it is reasonably straight forward to make recursive calls into `Make`, and use its own build ordering algorithms.
-
-The `unravel` tool outputs `./../build/pkg_artifacts/workplan.mk`
-
-#### Workplan Makefile
-`Unravel` creates a workplan which encodes both the ordering information and the build instructions for each package to be built. It creates a `.PHONY` target and recipe for every node in the graph. Many of these recipes are empty (i.e. `GOAL_PackagesToBuild: ;`), but any `TypeBuild` node with `StateBuild` will be given an actual build command invoking the `pkgworker` tool. The dependencies between nodes are encoded as dependencies between targets.
-
-A very simple workplan might look like:
-```makefile
-.PHONY: GOAL_PackagesToBuild
-GOAL_PackagesToBuild: ;
-GOAL_PackagesToBuild:  RUN_0_./../build/INTERMEDIATE_SRPMS/x86_64/example-1.0.0-1.cm1.src.rpm_example
-
-.PHONY: RUN_0_./../build/INTERMEDIATE_SRPMS/x86_64/example-1.0.0-1.cm1.src.rpm_example
-RUN_0_/home/damcilva/repos/demo/build/INTERMEDIATE_SRPMS/x86_64/example-1.0.0-1.cm1.src.rpm_example: ;
-RUN_0_/home/damcilva/repos/demo/build/INTERMEDIATE_SRPMS/x86_64/example-1.0.0-1.cm1.src.rpm_example:  BUILD_./../build/INTERMEDIATE_SRPMS/x86_64/example-1.0.0-1.cm1.src.rpm
-
-.PHONY: BUILD_./../build/INTERMEDIATE_SRPMS/x86_64/example-1.0.0-1.cm1.src.rpm
-BUILD_./../build/INTERMEDIATE_SRPMS/x86_64/example-1.0.0-1.cm1.src.rpm:
-	MAKEFLAGS= $(go-pkgworker) --input=./../build/INTERMEDIATE_SRPMS/x86_64/example-1.0.0-1.cm1.src.rpm --work-dir=$(CHROOT_DIR) ... >> $(LOGS_DIR)/pkggen/failures.txt 
-```
+`scheduler` controls a pool of build agents (`pkgworker`). It starts at the leaf nodes of the dependency graph and processes every node. Only processing `TypeBuild` nodes causes srpms to be built. Other node types are effectively NoOps, only processed to enforce dependency ordering.
 
-The entry point is `GOAL_PackagesToBuild`, which depends on the run node for the desired package (since run nodes indicate a package can be installed). Each run node will list its dependencies.
+`scheduler` will avoid building an srpm if it detects the package has already been built, and all of its dependencies were also prebuilt. If any dependencies of an SRPM needed to be built, then that SRPM will be built regardless.
```

Comment:
>any dependencies [](start = 141, length = 16)

Wouldn't we only care to rebuild if a dependency from `BuildRequires` was modified? This may lead to some confusing rebuilds, where a change to one package, which is a `Requires` for another triggers an avalanche of rebuilds, even though the build-time packages weren't changed.

Reply:
You are correct, that is what the scheduler does. I'll clarify the documentation